docs/topics/whatsnew-eap.md

```diff
@@ -67,19 +67,110 @@ the Kotlin team at JetBrains is focusing on improving the language with new feat
+
+With this new DSL, you can configure compiler options at the extension level for all targets and shared source sets 
+like `commonMain`, and at the target level for specific targets:
+
+```kotlin
+kotlin {
+    compilerOptions {
+        // Extension-level common compiler options that are used as defaults
+        // for all targets and shared source sets
+        allWarningsAsErrors.set(true)
+    }
+    jvm {
+        compilerOptions {
+            // Target-level JVM compiler options that are used as defaults
+            // for all compilations in this target
+            noJdk.set(true)
+        }
+    }
+}
+```
+
+The overall project configuration now has three layers.
+The highest is the extension level, then the target level,
+and the lowest is the compilation unit (which is usually a compilation task):
+
+![Kotlin compiler options levels](compiler-options-levels.svg){width=700}
+
+The settings at a higher level are used as a convention (default) for a lower level:
+
+* The values of extension compiler options are the default for target compiler options, including shared source sets,
+  like `commonMain`, `nativeMain`, and `commonTest`.
+* The values of target compiler options are used as the default for compilation unit (task) compiler options, for
+  example, `compileKotlinJvm` and `compileTestKotlinJvm` tasks.
+
+In turn, configurations made at a lower level override related settings at a higher level:
+
+* Task-level compiler options override related configurations at the target or the extension level.
+* Target-level compiler options override related configurations at the extension level.
+
+When configuring your project, keep in mind that some old ways
+of setting up compiler options have been [deprecated](whatsnew20.md#deprecated-old-ways-of-defining-compiler-options).
+
+## Improved overload resolution for functions with generic types
+
+Previously, if you had a number of overloads for a function where some have value parameters of generic type
+and others had function types at the same position, the resolution behavior was inconsistent in some cases.
+
+This led to different behavior depending on whether your overloads were member functions or extension functions.
+For example:
+
+```kotlin
+class KeyValueStore<K, V> {
+    fun store(key: K, value: V) {} // 1
+    fun store(key: K, lazyValue: () -> V) {} // 2
+}
+
+fun <K, V> KeyValueStore<K, V>.storeExtension(key: K, value: V) {} // 1 
+fun <K, V> KeyValueStore<K, V>.storeExtension(key: K, lazyValue: () -> V) {} // 2
+
+fun test(kvs: KeyValueStore<String, Int>) {
+    // Member functions
+    kvs.store("", 1)    // Resolves to 1
+    kvs.store("") { 1 } // Resolves to 2
+
+    // Extension functions
+    kvs.storeExtension("", 1)    // Resolves to 1
+    kvs.storeExtension("") { 1 } // Doesn't resolve
+}
+```
+
+In the example, the `KeyValueStore` class has two overloads for the `store()` function,
+where one overload has function parameters with generic types `K`, `V`,
```

Comment:
Maybe `<K, V>`  OR " `K` and `V`, "

Reply:
It’s more of a nitpick. I’d like to keep it as it is.